CommandLine/emake/Game.hpp

```diff
@@ -9,22 +9,33 @@
 class Game
 {
 public:
+  void AddSimpleObject(std::string create, std::string step, std::string draw);
+  void AddDefaultRoom();
   EnigmaStruct* ConstructGame();
   ~Game();
 
 protected:
-  EnigmaStruct* _game;
-  std::vector<Script*> _scripts;
-  std::vector<Sprite*> _sprites;
-  std::vector<Shader*> _shaders;
-  std::vector<Sound*> _sounds;
-  std::vector<Font*> _fonts;
-  std::vector<Background*> _backgrounds;
-  std::vector<Path*> _paths;
-  std::vector<Timeline*> _timelines;
-  std::vector<GmObject*> _objects;
-  std::vector<Room*> _rooms;
-  GameSettings* _gameSettings;
+  EnigmaStruct _game;
+  std::string _output_filename;
+  std::vector<Script> _scripts;
+  std::vector<Sprite> _sprites;
+  std::vector<Shader> _shaders;
+  std::vector<Sound> _sounds;
+  std::vector<Font> _fonts;
+  std::vector<Background> _backgrounds;
+  std::vector<Path> _paths;
+  std::vector<Timeline> _timelines;
+  std::vector<GmObject> _objects;
+  std::vector<Room> _rooms;
+
+  std::vector<std::vector<MainEvent>> _main_events;
+  std::vector<std::vector<Event>> _events;
+  std::vector<std::vector<Instance>> _instances;
+  std::vector<std::string> _strings;
+
+  const char* SS(std::string str);
+  void AddSimpleEvent(std::vector<MainEvent>* main_events,
```

Comment:
I believe supporting SOG would be better for testing overall so we can ensure events like collisions, animation end and game end are actually being triggered properly in our test cases. That said, I think the number of events in SOG should be optional as to not make it a pain to create one because even  I have no idea what half the events are named in GM. I'm willing to accept 3fg but I believe we would be wanting sog instead soon...

Reply:
The idea is that we will have EGM before that. However, even in the presence of EGM, perhaps SOG will be useful. I've implemented it.

Reply:
Agree here, lots of quick indie competitions also end up as SOG. So it's a bonus that it might also be able to run semi-real games! Theoretically speaking, a single object is all you actually ever need in order to create any game in GM. While it may be really difficult and messy, you could, hypothetically, make something like Iji with a single object.

Reply:
I think SOG would be cleaner than egm for adding tests to our source tree.